### PR TITLE
[NEW] Add Support for Salesforce URL Site Switching

### LINKED ISSUE #1302: [SF.app] Design and implement dynamic Salesforce URL site switching
We recently had an issue where we failed to escalate sessions because Salesforce itself did an update and switched our site URL without notifying us.  That is apparently normal Salesforce behavior, and part of their normal failover/maintenance. Here is an article about it, as well as how we are supposed to handle it:
 
https://help.salesforce.com/s/articleView?id=000337828&type=1
https://help.salesforce.com/s/articleView?id=000336101&type=1
 
TDLR: There’s an API endpoint we can hit that will send us a “SwitchServer” payload if we are using the wrong URL, that contains the new URL that we should be using:
 
https://developer.salesforce.com/docs/atlas.en-us.live_agent_rest.meta/live_agent_rest/live_agent_rest_implement_chat_visitor_experience_requests.htm
 
We need to design how we can use this endpoint to automatically identify, and update our SF site url in the case that it has changed.  We do expect this to happen at least a couple of times a year if not more frequently.

### PR DESCRIPTION
**Pull request by Shailesh351**
Closes https://github.com/WideChat/Rocket.Chat/issues/1302

For testing, use `https://la3-c1cs-chi.salesforceliveagent.com/chat/rest` as salesforce chat API in the SF App Setting

### CODE REVIEW

**Comment by ear-dev** [comment]
@Shailesh351 I've tested this against both, our sf_emulator and the sf url mentioned in the description. They seem to behave differently and I'm not yet sure what to make of it. 

- They both work. i.e. a successful escalation to our sf instances works.
- But the url mentioned above, does not seem to redirect, or receive the switchServer url, or throw the log point I'm looking for. 
- When testing against the emulator, it all does as expected...  I see the redirect, I get the log point etc..

I guess I'm not sure what mechanism that url above uses and if it represents what we will see in production?

**Comment by Shailesh351** [comment]
@ear-dev The salesforce URL I shared above seems region/instance specific. It's routing my SF instance to a new URL but for pre-prod it's sending availability payload.

Can you use `https://la4-c1cs-phx.salesforceliveagent.com/chat/rest` instead? I checked we are receiving `SwitchServer` with this URL.

Here's the curl for the same.

```
curl --location --request GET 'https://la4-c1cs-phx.salesforceliveagent.com/chat/rest/Visitor/Availability?org_id=00DP0000003p04l&deployment_id=572P00000004Csh&Availability.ids=573P00000004H5F' \
--header 'X-LIVEAGENT-API-VERSION: 49' \
--header 'Cookie: X-Salesforce-CHAT=!WT14m28CiR2kIiG8tLWddRD5lqVeAdHzeteajhwrPoiSDiw2qVesfoLbS/NfHPTL/F3LBhN5+bnR4c8='
```

As per the design, It will log an error for the SwitchServer and will still escalate the chat with a new URL.

You can find other salesforce URLs here https://help.salesforce.com/s/articleView?id=000387161&type=1, one of them should work I guess.

**Comment by ear-dev** [comment]
Nice, thanks @Shailesh351 I will test.  But like I said, with our emulator it worked great......  so looking good.  Thanks.

**Comment by ear-dev** [comment]
@bhardwajaditya I've tested this PR and it works great.  Can you please take a look at the code, do a review, and approve if it looks good to you?  Thanks!

File: handlers/SalesforceAgentAssignedHandler.ts [hunk 1]
```diff
@@ -31,15 +31,8 @@ export class SalesforceAgentAssigned {
...
 		const technicalDifficultyMessage: string = await getAppSettingValue(this.read, AppSettingId.TECHNICAL_DIFFICULTY_MESSAGE);
 
-		let salesforceChatApiEndpoint: string = await getAppSettingValue(this.read, AppSettingId.SALESFORCE_CHAT_API_ENDPOINT);
-		try {
-			salesforceChatApiEndpoint = salesforceChatApiEndpoint.replace(/\/?$/, '/');
-		} catch (error) {
-			await sendLCMessage(this.read, this.modify, this.data.room, technicalDifficultyMessage, this.data.agent);
-			await sendDebugLCMessage(this.read, this.modify, this.data.room, ErrorLogs.SALESFORCE_CHAT_API_NOT_FOUND, this.data.agent);
-			console.error(ErrorLogs.SALESFORCE_CHAT_API_NOT_FOUND, error);
-			return;
```
Comment (bhardwajaditya) [inline handlers/SalesforceAgentAssignedHandler.ts:41]: We are not sending technical difficulty message anymore post this change. Is this expected behaviour?

Comment (Shailesh351) [reply handlers/SalesforceAgentAssignedHandler.ts:41]: Yes, we are not sending technicalDifficultyMessage. It's redundant. In the try-catch we are only doing string replacement. Also, we are not having access to all the parameters of sendMessage fn everywhere.

**Review by bhardwajaditya — APPROVED** [review]
If it's by design then no issues. Rest all looks good to me

**Pull request merged**